libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Trace.hs

```diff
@@ -344,41 +319,38 @@ badScripts proof xs = Fold.foldl' (\s mcf -> Set.union s (getw proof mcf)) Set.e
         ) = set
     getw _ _ = Set.empty
 
-showBlock :: forall era. Reflect era => MUtxo era -> [Tx era] -> PDoc
+showBlock :: forall era. (Reflect era, EraTest era) => MUtxo era -> [Tx era] -> Expr
 showBlock u txs = ppList pppair (zip txs [0 ..])
   where
     pppair (tx, n) =
       let body = getBody reify tx
-       in vsep
-            [ ppString "\n###########"
-            , ppInt n
-            , ppSafeHash (hashAnnotated body)
-            , smartTxBody reify u body
-            , ppString (show (isValid' reify tx))
-            , ppMap pcScriptHash (pcScript @era reify) (getScriptWits reify (getWitnesses reify tx))
-            , ppString "\n"
+       in toExpr
+            [ toExpr @Int n
+            , toExpr $ hashAnnotated body
+            , smartTxBody u body
+            , toExpr (isValid' reify tx)
+            , ppMap toExpr toExpr (getScriptWits reify (getWitnesses reify tx))
             ]
 
-shortTxOut :: EraTxOut era => Proof era -> TxOut era -> PDoc
+shortTxOut :: EraTxOut era => Proof era -> TxOut era -> Expr
 shortTxOut proof out = case txoutFields proof out of
-  (Addr _ pay _, _, _) ->
-    hsep ["Out", parens $ hsep ["Addr", pcCredential pay], pcCoin (out ^. coinTxOutL)]
+  (Addr _ pay _, _, _) -> toExpr (pay, out ^. coinTxOutL)
   _ -> error "Bootstrap Address in shortTxOut"
 
-smartTxBody :: Reflect era => Proof era -> MUtxo era -> TxBody era -> PDoc
-smartTxBody proof u txbody = ppRecord "TxBody" pairs
-  where
-    fields = abstractTxBody proof txbody
-    pairs = concat (map help fields)
-    help (Inputs s) = [("spend inputs", ppSet pcIn s)]
-    help (Collateral s) = [("coll inputs", ppSet pcIn s)]
-    help x = pcTxBodyField proof x
-    pcIn x =
-      hsep
-        [ pcTxIn x
-        , " -> "
-        , case Map.lookup x u of Just out -> shortTxOut proof out; Nothing -> "?"
-        ]
+smartTxBody :: EraTest era => MUtxo era -> TxBody era -> Expr
+smartTxBody u txbody = toExpr (u, txbody) -- ppRecord "TxBody" pairs
+-- where
+--   fields = abstractTxBody proof txbody
+--   pairs = concat (map help fields)
+--   help (Inputs s) = [("spend inputs", ppSet pcIn s)]
+--   help (Collateral s) = [("coll inputs", ppSet pcIn s)]
+--   help x = pcTxBodyField proof x
+--   pcIn x =
+--     hsep
+--       [ pcTxIn x
+--       , " -> "
+--       , case Map.lookup x u of Just out -> shortTxOut proof out; Nothing -> "?"
+--       ]
```

Comment:
Don't worry about keeping the prettiness. What you have here is just fine, especially considering that we wanna get rid of this stuff eventually anyways.
```suggestion
smartTxBody u txbody = toExpr (u, txbody)
```